### PR TITLE
initial buildout of dns guide

### PR DESCRIPTION
**Pull request by uzair-ashraf**

### CODE REVIEW

**Comment by TimelikeClosure** [comment]
Marked as a draft until the missing hover.com screenshots are added in

**Comment by TimelikeClosure** [comment]
@uzair-ashraf Don't forget to throw a link up on the main `README.md`

**Comment by uzair-ashraf** [comment]
@TimelikeClosure Which section should I put it in?

**Comment by TimelikeClosure** [comment]
> @TimelikeClosure Which section should I put it in?

@uzair-ashraf Probably after the EC2 Initial Setup (since they need the IP address), but before the NGINX setup (since DNS records need time to propagate through the DNS network).

File: DNS_SETUP.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 This guide will walk you through setting up your DNS for your domain name.
 
+Upon completion of setting up your DNS Settings checkout the following guides to finish your deployment.
+
+- [Deploying a Full Stack Application](https://github.com/Learning-Fuze/full-stack-project/blob/master/guides/DEPLOYMENT.md)
```
Comment (TimelikeClosure) [inline DNS_SETUP.md:7]: Might cause some issues if they get access to this repo well before they get access to the final project repo, but it should be fine as long as we warn them ahead of time.

File: DNS_SETUP.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
...
+- [Deploying a Static Application](https://github.com/Learning-Fuze/memory_match/tree/master/guides/deployment)
```
Comment (TimelikeClosure) [inline DNS_SETUP.md:8]: We might want to re-locate this guide, since `memory_match` is not part of the portfolio requirements anymore. I suspect moving it into this repo might work. Regardless, this is outside the scope of this PR.

Comment (TimelikeClosure) [reply DNS_SETUP.md:8]: Also, apparently at least one of the Part Time Module classes never got access to that repo.

File: DNS_SETUP.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
+# DNS Setup
+
+This guide will walk you through setting up your DNS for your domain name.
```
Comment (thebearingedge) [inline DNS_SETUP.md:3]: This guide will walk you through setting up DNS records for your domain name.

File: DNS_SETUP.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Upon completion of setting up your DNS Settings checkout the following guides to finish your deployment.
```
Comment (thebearingedge) [inline DNS_SETUP.md:5]: Upon completion of adding your DNS records, check out the following guides to finish your deployments.

File: DNS_SETUP.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+- [Deploying a Full Stack Application](https://github.com/Learning-Fuze/full-stack-project/blob/master/guides/DEPLOYMENT.md)
+- [Deploying a Static Application](https://github.com/Learning-Fuze/memory_match/tree/master/guides/deployment)
+
+## Adding an A Record
+
+A Records are the most basic type of DNS record and are used to point a domain or subdomain to an IP address.  You only need to do this the first time you are setting up your EC2 instance.  Once you have received your elastic ip address you must setup an `A Record` to have your domain name service point to ip address where your nginx server is hosted.
```
Comment (thebearingedge) [inline DNS_SETUP.md:12]: > Once you have received your elastic ip address you must setup an...

Once you have allocated an Elastic IP address to your EC2 instance, you must add an...

File: DNS_SETUP.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+
+Setting up an A Record on:
```
Comment (thebearingedge) [inline DNS_SETUP.md:15]: Adding an A Record on:

File: DNS_SETUP.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+- [namecheap.com](#setting-an-a-record-on-namecheap.com)
+- [name.com](#setting-an-a-record-on-name.com)
+- [hover.com](#setting-an-a-record-on-hover.com)
+
+
+### Setting an A Record on namecheap.com
+
+1. Login to namecheap.com and land on account page
```
Comment (thebearingedge) [inline DNS_SETUP.md:23]: Log in

File: DNS_SETUP.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Click `Manage` on the domain name that you want to point to your server.
```
Comment (thebearingedge) [inline DNS_SETUP.md:25]: > point to your server.

point to your Elastic IP address.

File: AWS_EC2_INITIAL_SETUP.md [hunk 9]
```diff
@@ -49,7 +49,7 @@ This guide will walk you through setting up an EC2 (Free Tier) Instance on AWS.
     # ![Review Instance Launch](images/aws_ec2_initial_setup/review-instance-launch.png)
 1. Once you have ensured everything is correct, click the blue `Launch` button in the bottom right corner.
     # ![Click blue launch button](images/aws_ec2_initial_setup/launch.png)
-1. You should see a modal pop up that says **"Select an existing key pair or create a new key pair"**. 
+1. You should see a modal pop up that says **"Select an existing key pair or create a new key pair"**.
     ### THESE STEPS ARE VERY IMPORTANT!! Without this file you can not access your server!
```
Comment (thebearingedge) [inline AWS_EC2_INITIAL_SETUP.md:53]: > Without this file you can not access your server!

Without this file you can not access your EC2 instance.

File: DNS_SETUP.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Click `Add New Record`
+
+    ![namecheap add new record](images/dns_setup/namecheap-3.png)
+
+1. Click the green check mark once you have added the following fields
+
+    1. Set the first drop down to `A Record`
+    1. The second field is `@` (the @ symbol represents your root domain.)
+    1. The third field is the elastic ip address you got from aws.
```
Comment (thebearingedge) [inline DNS_SETUP.md:41]: >     1. The third field is the elastic ip address you got from aws.

    1. The third field is the Elastic IP address you allocated to your EC2 instance.

File: DNS_SETUP.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+    ![namecheap domain manage](images/dns_setup/namecheap-4.png)
+
+    [Setting a CNAME record on namecheap.com](#setting-a-cname-record-on-namecheap.com)
+
+
+### Setting an A Record on name.com
+
+1. Login to name.com and land on the account page
```
Comment (thebearingedge) [inline DNS_SETUP.md:50]: Log in

File: DNS_SETUP.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+    1. Set the first drop down to `A Record`
+    1. The second field is `@` (the @ symbol represents your root domain.)
+    1. The third field is the elastic ip address you got from aws.
+
+    ![namecheap domain manage](images/dns_setup/namecheap-4.png)
+
+    [Setting a CNAME record on namecheap.com](#setting-a-cname-record-on-namecheap.com)
+
+
+### Setting an A Record on name.com
```
Comment (thebearingedge) [inline DNS_SETUP.md:48]: > ### Setting an A Record on name.com

### Adding an A Record on name.com

File: DNS_SETUP.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Login to name.com and land on the account page
+
+1. Click on the domain name that you want to point to your server.
```
Comment (thebearingedge) [inline DNS_SETUP.md:52]: > 1. Click on the domain name that you want to point to your server.

1. Click on the domain name that you want to point to your Elastic IP address.

File: DNS_SETUP.md [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Click on `Manage DNS Records`
+
+    ![name domain manage](images/dns_setup/name-2.png)
+
+1. Click the blue `Add Record` button once you have added the following fields
+
+    1. Set the first drop down to `A Record`
+    1. The second field is your domain name
+    1. The third field is the elastic ip address you got from aws.
```
Comment (thebearingedge) [inline DNS_SETUP.md:64]: >     1. The third field is the elastic ip address you got from aws.

    1. The third field is the Elastic IP address you got from AWS.

File: DNS_SETUP.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+    ![name domain manage](images/dns_setup/name-3.png)
+
+    [Setting a CNAME record on name.com](#setting-a-cname-record-on-name.com)
+
+
+### Setting an A Record on hover.com
+
+1. Login to hover.com and land on the account page
```
Comment (thebearingedge) [inline DNS_SETUP.md:73]: Log in

File: DNS_SETUP.md [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Click on the edit drop down of the domain name that you want to point to your server.
```
Comment (thebearingedge) [inline DNS_SETUP.md:75]: > 1. Click on the domain name that you want to point to your server.

1. Click on the domain name that you want to point to your Elastic IP address.

File: DNS_SETUP.md [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Click the `Add A Record` button.
+
+    ![hover domain manage](images/dns_setup/hover-3.png)
+
+1. Click the `Add Record` after inputting the following data
+
+    1. Set the `TYPE` drop down to `A`
+    1. Set `HOSTNAME` to your domain name
+    1. Set `IP ADDRESS` to the elastic ip address you got from aws.
```
Comment (thebearingedge) [inline DNS_SETUP.md:91]: Elastic IP address you got from AWS

File: DNS_SETUP.md [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+    ![hover domain manage](images/dns_setup/hover-4.png)
+
+    [Setting a CNAME record on hover.com](#setting-a-cname-record-on-hover.com)
+
+
+___
+
+## Adding a CNAME Record
+
+The DNS CNAME record works as an alias for domain names that share a single IP address.  CNAME records tell the subdomain to use the same IP address as your A record. With multiple web applications, you create multiple CNAME records so if you ever have to change your IP address, you only have to do it once. Every time you want to deploy a web application for your portfolio, you will need to do this.
```
Comment (thebearingedge) [inline DNS_SETUP.md:102]: > CNAME records tell the subdomain to use the same IP address as your A record.

CNAME records tell DNS resolvers to use the same IP address as your A record.
